### ISSUE TITLE
ffmpeg libavcodec/libx264 make error 'x264_bit_depth' undeclared

### ISSUE THREAD

**Issue by BlohoJo** [jb-alvarado/media-autobuild_suite#675]
**_Edit - Error is caused by [changes to x264 version/libraries](https://trac.ffmpeg.org/ticket/6932), **NOT** by xpcomp=1, see replies below._**

Trying to get a 32 bit binary build of the suite that will run on our XP computers.

Running clean / from scratch with latest media-autobuild_suite (timestamp 12/25/2017 4:32 PM EST)

Using .ini settings arch=2 (Build 32 bit only), xpcomp=1 (Build with XP compatibility when possible), x2652=1, and mpv=2.

~~I'm not sure if this error is expected with the .ini settings I'm using or not. If it is, could you please let me know what to turn off/on to get a XP compatible build?~~

~~Of course, I didn't try the _non_ XP settings that I used the last time I ran the suite successfully (timestamp 12/24/17 12:38 PM), so this _could_ be a new bug with the newer version of media-autobuild_suite. I'll check and report back.~~

.ini:
```
[compiler list]
msys2Arch=2
arch=2
license2=1
xpcomp=1
standalone=1
vpx2=1
aom=1
daala=1
x2643=4
x2652=1
other265=1
flac=1
fdkaac=1
faac=1
mediainfo=1
soxB=1
ffmpegB2=4
ffmpegUpdate=1
ffmpegChoice=4
mp4box=1
rtmpdump=1
mplayer2=1
mpv=2
bmx=1
curl=1
ffmbc=1
cyanrip=1
cores=3
deleteSource=1
strip=1
pack=2
logging=1
updateSuite=1
```

Error:
```
┌ ffmpeg git ........................................ [Recently updated]
├ Changing options to comply to nonfree...
├ Compiling shared FFmpeg...
├ Running configure...
├ Running make...
Likely error:
     if (x264_bit_depth > 8)
         ^~~~~~~~~~~~~~
         x264_picture_t
C:/mingw/build/ffmpeg-git/libavcodec/libx264.c:282:9: note: each undeclared identifier is reported only once for each function it appears in
C:/mingw/build/ffmpeg-git/libavcodec/libx264.c: In function 'X264_init_static':
C:/mingw/build/ffmpeg-git/libavcodec/libx264.c:876:9: error: 'x264_bit_depth' undeclared (first use in this function); did you mean 'x264_picture_t'?
     if (x264_bit_depth == 8)
         ^~~~~~~~~~~~~~
         x264_picture_t
make: *** [/build/ffmpeg-git/ffbuild/common.mak:60: libavcodec/libx264.o] Error 1
make failed. Check C:/mingw/build/ffmpeg-git/build-shared-32bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

Attach C:\mingw\build\logs.zip to the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
Try running the build again at a later time.
```

[/build/ffmpeg-git/build-shared-32bit/ab-suite.configure.log](https://github.com/jb-alvarado/media-autobuild_suite/files/1586610/ab-suite.configure.log)
[/build/ffmpeg-git/build-shared-32bit/ab-suite.make.log](https://github.com/jb-alvarado/media-autobuild_suite/files/1586611/ab-suite.make.log)

[/build/logs.zip](https://github.com/jb-alvarado/media-autobuild_suite/files/1586612/logs.zip)

**Comment by schmidthubert** [comment]
For recent ffmpeg XP changes read comment here: https://git.ffmpeg.org/gitweb/ffmpeg.git/commitdiff/9b121dfc32810250938021952aab4172a988cb56

Apart from that the x264 version/libs have very recently changed to x264 0.155.2893 b00bcaf which will cause another hiccup for several days not only on here but everywhere where that code is used .... 

See hydra3333's [ticket](https://trac.ffmpeg.org/ticket/6932) as well and wait for Riccardo to fix up the scripts here .... :+1:

**Comment by GyanD** [comment]
FFmpeg has dropped XP support

http://git.videolan.org/?p=ffmpeg.git;a=commit;h=9b121dfc32810250938021952aab4172a988cb56

**Comment by BlohoJo** [comment]
Yeah, I checked again with the non-xp build and I get the same error starting clean / from scratch. This error is **NOT** caused by writexpcomp=yes. It's related to the other new issue #676.

.ini:
```
[compiler list]
msys2Arch=2
arch=1
license2=1
xpcomp=2
standalone=1
vpx2=1
aom=1
daala=1
x2643=4
x2652=6
other265=1
flac=1
fdkaac=1
faac=1
mediainfo=1
soxB=1
ffmpegB2=4
ffmpegUpdate=1
ffmpegChoice=4
mp4box=1
rtmpdump=1
mplayer2=1
mpv=1
bmx=1
curl=1
ffmbc=1
cyanrip=1
cores=3
deleteSource=1
strip=1
pack=2
logging=1
updateSuite=1
```


Error:
```
┌ ffmpeg git ........................................ [Recently updated]
├ Changing options to comply to nonfree...
├ Compiling shared FFmpeg...
├ Running configure...
├ Running make...
Likely error:
     if (x264_bit_depth > 8)
         ^~~~~~~~~~~~~~
         x264_picture_t
C:/mingw/build/ffmpeg-git/libavcodec/libx264.c:282:9: note: each undeclared identifier is reported only once for each function it appears in
C:/mingw/build/ffmpeg-git/libavcodec/libx264.c: In function 'X264_init_static':
C:/mingw/build/ffmpeg-git/libavcodec/libx264.c:876:9: error: 'x264_bit_depth' undeclared (first use in this function); did you mean 'x264_picture_t'?
     if (x264_bit_depth == 8)
         ^~~~~~~~~~~~~~
         x264_picture_t
make: *** [/build/ffmpeg-git/ffbuild/common.mak:60: libavcodec/libx264.o] Error 1
make failed. Check C:/mingw/build/ffmpeg-git/build-shared-32bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/s8tP.zip
Copy and paste [logs.zip](https://0x0.st/s8tP.zip) in the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
Try running the build again at a later time.
```

[/build/ffmpeg-git/build-shared-32bit/ab-suite.configure.log](https://github.com/jb-alvarado/media-autobuild_suite/files/1586893/ab-suite.configure.log)
[/build/ffmpeg-git/build-shared-32bit/ab-suite.make.log](https://github.com/jb-alvarado/media-autobuild_suite/files/1586894/ab-suite.make.log)

[/build/logs.zip](https://github.com/jb-alvarado/media-autobuild_suite/files/1586895/logs.zip)

**Comment by BlohoJo** [comment]
> _schmidthubert commented:_
> For recent ffmpeg XP changes read comment here: https://git.ffmpeg.org/gitweb/ffmpeg.git/commitdiff/9b121dfc32810250938021952aab4172a988cb56

> _GyanD commented:_
> FFmpeg has dropped XP support
> 
> http://git.videolan.org/?p=ffmpeg.git;a=commit;h=9b121dfc32810250938021952aab4172a988cb56

Yes, I know, but media-autobuild_suite has an XP Compatibility switch (xpcomp=1). I figured that if it's still there it's supported; I assumed it was doing something similar to [Reino17's legacy XP/Non-SSE2 builds](https://github.com/rdp/ffmpeg-windows-build-helpers/issues/219#issuecomment-314933108), or was using his [published build helpers](https://github.com/Reino17/ffmpeg-windows-build-helpers/commit/a7f5cd5db3716eb9e8362d54f3dc4c6265801679). If XP Compatibility isn't possible anymore with media-autobuild_suite maybe it should be removed to stop dummies like me who still use XP computers from posting issues about it. :)

**Comment by wiiaboo** [comment]
The x264 is being fixed upstream. Just wait.

And yeah, XP support is gone. I'm removing xpcomp. Good riddance.

**Comment by BlohoJo** [comment]
lol ok :)

**Comment by wiiaboo** [comment]
Fixed upstream.
